packages/block-library/src/embed/edit.js

```diff
@@ -0,0 +1,382 @@
+		 * if the HTML has an iframe with width and height set.
+		 *
+		 * @param {string} html The preview HTML that possibly contains an iframe with width and height set.
+		 * @param {boolean} allowResponsive If the classes should be added, or removed.
+		 * @return {Object} Object with classnames set for use with `classnames`.
+		 */
+		getAspectRatioClassNames( html, allowResponsive = true ) {
+			const previewDocument = document.implementation.createHTMLDocument( '' );
+			previewDocument.body.innerHTML = html;
+			const iframe = previewDocument.body.querySelector( 'iframe' );
+
+			if ( iframe && iframe.height && iframe.width ) {
+				const aspectRatio = ( iframe.width / iframe.height ).toFixed( 2 );
+				// Given the actual aspect ratio, find the widest ratio to support it.
+				for ( let ratioIndex = 0; ratioIndex < ASPECT_RATIOS.length; ratioIndex++ ) {
+					const potentialRatio = ASPECT_RATIOS[ ratioIndex ];
+					if ( aspectRatio >= potentialRatio.ratio ) {
+						return {
+							[ potentialRatio.className ]: allowResponsive,
+							'wp-has-aspect-ratio': allowResponsive,
+						};
+					}
+				}
+			}
+
+			return this.props.attributes.className;
+		}
+
+		/**
+		 * Sets the aspect ratio related class names returned by `getAspectRatioClassNames`
+		 * if `allowResponsive` is truthy.
+		 *
+		 * @param {string} html The preview HTML.
+		 */
+		setAspectRatioClassNames( html ) {
+			const { allowResponsive } = this.props.attributes;
+			if ( ! allowResponsive ) {
+				return;
+			}
+			const className = classnames(
+				this.props.attributes.className,
+				this.getAspectRatioClassNames( html )
+			);
+			this.props.setAttributes( { className } );
+		}
+
+		/***
+		 * Gets block attributes based on the preview and responsive state.
+		 *
+		 * @param {string} preview The preview data.
+		 * @param {boolean} allowResponsive Apply responsive classes to fixed size content.
+		 * @return {Object} Attributes and values.
+		 */
+		getAttributesFromPreview( preview, allowResponsive = true ) {
+			const attributes = {};
+			// Some plugins only return HTML with no type info, so default this to 'rich'.
+			let { type = 'rich' } = preview;
+			// If we got a provider name from the API, use it for the slug, otherwise we use the title,
+			// because not all embed code gives us a provider name.
+			const { html, provider_name: providerName } = preview;
+			const providerNameSlug = kebabCase( toLower( '' !== providerName ? providerName : title ) );
+
+			if ( isFromWordPress( html ) ) {
+				type = 'wp-embed';
+			}
+
+			if ( html || 'photo' === type ) {
+				attributes.type = type;
+				attributes.providerNameSlug = providerNameSlug;
+			}
+
+			attributes.className = classnames(
+				this.props.attributes.className,
+				this.getAspectRatioClassNames( html, allowResponsive )
+			);
+
+			return attributes;
+		}
+
+		/***
+		 * Sets block attributes based on the preview data.
+		 */
+		setAttributesFromPreview() {
```

Comment:
setAttributesFromPreview seems to only be called in one place, so could be inlined.